apps/docs/stories/architecture-decision-records/0007-why-do-we-transpile-web-components-for-react-and-vue.stories.mdx

```diff
@@ -0,0 +1,40 @@
+import { Meta } from '@storybook/addon-docs'
+
+<Meta title='Docs/Architecture Decision Records/ADR 0007: Why do we transpile Web Components for React and Vue?' />
+
+# ADR 0007: Why do we transpile Web Components for React and Vue?
+
+🗓️ ~2023-04~ 2024-05 · ✍️ [@felipefialho](https://twitter.com/felipefialho_)
+
+## Context
+
+We need to use Web Components in our projects built with Vue and React (Next.js) while maintaining a good Developer Experience (DX).
+
+## Problems
+
+### React
+
+- We need to use `Event Listeners` to handle events and methods when using Web Components
+- React [use `SyntheticEvent` to handle events](https://legacy.reactjs.org/docs/events.html), and we need to convert it to `CustomEvent` to use Web Components
+- Props that use `array` or `object` need to be converted to JSON for compatibility
+- We need to use `ref` to access the Web Component API in React
+- Sometimes, we need to create wrappers to effectively use Web Components in React
+- Overall, the Developer Experience in React is not optimal when working with Web Components
```

Comment:
:/

Reply:
@tassioFront React@19 will have a better experience (https://custom-elements-everywhere.com/)

<img width="1470" alt="Captura de Tela 2024-05-15 às 09 29 17" src="https://github.com/juntossomosmais/atomium/assets/54173994/b5239399-c008-4fe6-af84-8faaad7a1fbc">

Reply:
What a nice news!!! 🚀 ❤️

Reply:
But, @igorwessel I think that our experience using transpiler It's been good, don't you think?

Reply:
It's great!

I just wanted to show that perhaps the experience without the wrapper will become better, even removing that fix we have for React. 😃

Reply:
Yes! We will use the wrapper in our case, so it is not a problem for us. The advantage is that the Stencil wrapper probably will improve after this update